src/Game.js

```diff
@@ -0,0 +1,42 @@
+function Game() {
+  this.frames = [];
+  this.totalScore = 0;
+}
+
+Game.prototype.addFrame = function addFrame(frame) {
+  this.frames.push(frame);
+};
+
+Game.prototype.finalScore = function finalScore() {
```

Comment:
As you said in your readme, this function is probably doing too much at the moment.  It would benefit from some refactoring and breaking out into some smaller functions.  It also has lots of nested if/else logic which makes it a bit difficult to follow.